csi/node_server.go

```diff
@@ -647,6 +647,11 @@ func (ns *NodeServer) getMounter(volume *longhornclient.Volume, volumeCapability
 			params += fsParams.formatParameters
 		}
 
+		// If the user specifies parameters in the storage class, the parameters are overridden.
+		if mkfsParams, ok := volumeContext["mkfsParams"]; ok && mkfsParams != "" {
+			params = mkfsParams
```

Comment:
Should append instead of overwriting, because some default params of the supported fs are mandatory. 

Also, because of https://github.com/longhorn/longhorn/issues/4594 introduced, we only allow using the fixed block size (4096) which is defined in `supportedFs`, so we should filter it if it is specified in the user provided `mkfsParams`.

Reply:
Of course, we could allow users to do that w/o any filtering or validation and they need to be responsible for that, but probably there would be some concerns later.

@derekbit @PhanLe1010 @weizhe0422  WDYT?

Reply:
I think `append` is better. Then, the later one can override the previous value.

Reply:
yeah. @derekbit do you have any concerns if users use non-4096 block size which is what we fixed/improved in https://github.com/longhorn/longhorn/issues/4594.

Reply:
No, only performance drop if use a smaller block size. We can mention it in the doc.

Reply:
Got it, I will modify it as 'params+=mkfsParams' to append user specified parameters.